apps/judicial-system/backend/src/app/modules/case-table/whereOptions/prosecutor.ts

```diff
@@ -0,0 +1,101 @@
+import { fn, Op } from 'sequelize'
+
+import {
+  CaseAppealState,
+  CaseState,
+  completedIndictmentCaseStates,
+  completedRequestCaseStates,
+  indictmentCases,
+  investigationCases,
+  restrictionCases,
+  type User as TUser,
+} from '@island.is/judicial-system/types'
+
+const prosecutorSharedWhereOptions = (user: TUser) => ({
+  [Op.or]: [
+    { creating_prosecutor_id: user.id },
+    { prosecutor_id: user.id },
+    { prosecutors_office_id: user.institution?.id },
+    { shared_with_prosecutors_office_id: user.institution?.id },
```

Comment:
The shared with prosecutors office id only applies to request cases. Please consider refactorying.